File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/db/DBSubTeclas.java
package com.valleapp.vallecom.db;

import android.annotation.SuppressLint;
import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;


/**
 * Created by valle on 13/10/14.
 */
public class DBSubTeclas extends DBBase {


    public DBSubTeclas(Context context) {
        super(context, "subteclas");
    }

    public void onCreate(SQLiteDatabase db) {
        db.execSQL("CREATE TABLE IF NOT EXISTS subteclas (ID INTEGER PRIMARY KEY, " +
                " Nombre TEXT, Incremento DOUBLE, IDTecla INTEGER," +
                " descripcion_t TEXT, descripcion_r TEXT)");
    }


    public JSONArray getAll(String id)
    {
        return filter("IDTecla="+id );
    }


    @SuppressLint("Range")
    @Override
    protected JSONObject cursorToJSON(Cursor res) {
        JSONObject obj = new JSONObject();
        try {
            obj.put("Nombre", res.getString(res.getColumnIndex("Nombre")));
            obj.put("Incremento", res.getString(res.getColumnIndex("Incremento")));
            obj.put("descripcion_t", res.getString(res.getColumnIndex("descripcion_t")));
            obj.put("descripcion_r", res.getString(res.getColumnIndex("descripcion_r")));
        }catch (Exception e){
            e.printStackTrace();
        }
        return  obj;
    }

    @Override
    protected ContentValues caragarValues(JSONObject o) {
        ContentValues values = new ContentValues();
        try {
            values.put("ID", o.getInt("id"));
            values.put("Nombre", o.getString("nombre"));
            values.put("Incremento",o.getString("incremento"));
            values.put("IDTecla", o.getString("tecla"));
            values.put("descripcion_r", o.getString("descripcion_r"));
            values.put("descripcion_t", o.getString("descripcion_t"));
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return values;
    }
}

File: android/ValleTPV/ValleTPVLib/src/main/java/com/valleapp/valletpvlib/db/DBSubTeclas.kt
package com.valleapp.valletpvlib.db

import android.annotation.SuppressLint
import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import org.json.JSONArray
import org.json.JSONObject

/**
 * Created by valle on 13/10/14.
 */
abstract class DBSubTeclas(context: Context?) : DBBase(context, "subteclas") {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(
            "CREATE TABLE IF NOT EXISTS subteclas (ID INTEGER PRIMARY KEY, " +
                    " Nombre TEXT, Incremento DOUBLE, IDTecla INTEGER," +
                    " descripcion_t TEXT, descripcion_r TEXT)"
        )
    }

    fun getAll(id: String): JSONArray {
        return filter("IDTecla=$id")
    }

    override fun inicializar() {
        val db = this.writableDatabase
        onCreate(db)
    }

    @SuppressLint("Range")
    override fun cursorToJSON(res: Cursor?): JSONObject? {
        val obj = JSONObject()
        try {
            obj.put("ID", res!!.getString(res.getColumnIndex("ID")))
            obj.put("tecla", res.getString(res.getColumnIndex("IDTecla")))
            obj.put("Nombre", res.getString(res.getColumnIndex("Nombre")))
            obj.put("Incremento", res.getString(res.getColumnIndex("Incremento")))
            obj.put("descripcion_t", res.getString(res.getColumnIndex("descripcion_t")))
            obj.put("descripcion_r", res.getString(res.getColumnIndex("descripcion_r")))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return obj
    }

    override fun caragarValues(o: JSONObject?): ContentValues {
        val values = ContentValues()
        try {
            values.put("ID", o!!.getInt("id"))
            values.put("Nombre", o.getString("nombre"))
            values.put("Incremento", o.getString("incremento"))
            values.put("IDTecla", o.getString("tecla"))
            values.put("descripcion_r", o.getString("descripcion_r"))
            values.put("descripcion_t", o.getString("descripcion_t"))
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return values
    }
}